MakingItBetterG55/src/application/MainSceneHandler.java

```diff
@@ -1,69 +1,235 @@
+			if (event.getCode() == KeyCode.Q && canMove == true) {
+
+				currentPlayerIndex[0] = (currentPlayerIndex[0]+1)%4;
+				HBox newBottomPlayerBox = createPlayerPanel(allPlayers[currentPlayerIndex[0]]);
+				VBox newRightPlayerBox = createPlayerPanelSides(allPlayers[(currentPlayerIndex[0] + 1) % 4]);
+				HBox newTopPlayerBox = createPlayerPanel(allPlayers[(currentPlayerIndex[0] + 2) % 4]);
+				VBox newLeftPlayerBox = createPlayerPanelSides(allPlayers[(currentPlayerIndex[0] + 3) % 4]);
+
+				formatBottomPanel(newBottomPlayerBox, root);
+				formatRightPanel(newRightPlayerBox, root);
+				formatTopPanel(newTopPlayerBox, root);
+				formatLeftPanel(newLeftPlayerBox,root);
+
+			}
+		});
+
 		return mainScene;
 	}
+
+	private void formatLeftPanel(VBox box, BorderPane root){
+		VBox.setVgrow(root, Priority.ALWAYS);
+		box.setMaxHeight(500);
+		box.setPrefWidth(70);
+		root.setLeft(box);
+		BorderPane.setAlignment(box, Pos.CENTER_LEFT);
+	}
+
+	private void formatRightPanel(VBox box, BorderPane root){
+		VBox.setVgrow(root, Priority.ALWAYS);
+		box.setMaxHeight(500);
+		box.setPrefWidth(70);
+		root.setRight(box);
+		BorderPane.setAlignment(box, Pos.CENTER_RIGHT);
+	}
+
+	private void formatTopPanel(HBox box, BorderPane root){
+		box.setPrefHeight(60);
+		box.setMaxWidth(500);
+		root.setTop(box);
+		BorderPane.setAlignment(box, Pos.TOP_CENTER);
+	}
+
+	private void formatBottomPanel(HBox box, BorderPane root){
+		HBox.setHgrow(root, Priority.ALWAYS);
+		box.setMaxWidth(1000);
+		box.setPrefHeight(100);
+		root.setBottom(box);
+		BorderPane.setAlignment(box,Pos.CENTER);
+	}
+
+	private HBox createPlayerPanel(Player currentPlayer){
+
+		String backgroundColor = getPlayerColor(currentPlayer.getPlayerType());
+		String borderColor = getBorderColor(currentPlayer.getPlayerType());
+
+		//creates a horizontal box (for the top/bottom of the screen
+		HBox playerPanel = new HBox();
+		playerPanel.setStyle("-fx-background-color: " +  backgroundColor + ";-fx-border-color: " + borderColor + ";-fx-border-width: 5px;");
+
+
+		playerPanel.setPadding(new Insets(5));
+
+		//creates a label for each player resource
+		Label nameLabel = new Label(currentPlayer.getPlayerName());
+		Label scoreLabel = new Label(Integer.toString(currentPlayer.getscore()));
+		Label moneyLabel = new Label(Integer.toString(currentPlayer.getMoney()));
+		Label goodsLabel = new Label(Integer.toString(currentPlayer.getGoods()));
+		Label educationsLabel = new Label(Integer.toString(currentPlayer.getEducation()));
+		Label healthLabel = new Label(Integer.toString(currentPlayer.getHealth()));
+
+		//adds all of the player resources to the player panel box
+		playerPanel.getChildren().addAll(nameLabel, scoreLabel, moneyLabel, goodsLabel, educationsLabel, healthLabel);
+
+		return playerPanel;
+	}
+
+	private VBox createPlayerPanelSides(Player currentPlayer){
+
+		String backgroundColor = getPlayerColor(currentPlayer.getPlayerType());
```

Comment:
could just use a string array with length 8 containing all the colors then have array[turnNumber % 4] = color and array[(turnNumber % 4) *2] equal border color since you know when turnNumber % 4 == 0 it's the engineer, then 1 is teacher... etc. this would eleminate those two getColor methods

Reply:
I couldn't quite get this to work, fixing the main conflicts took a while so I might move on to my next ticket for now if that's ok :)